### ISSUE TITLE
Push / Pull very slow on all repositories

### ISSUE THREAD

**Issue by zaphbbrox** [go-gitea/gitea#19053]
### Gitea Version

Gitea version 1.16.3 built with GNU Make 4.1, go1.17.7 : bindata, sqlite, sqlite_unlock_notify

### Git Version

git version 2.20.1

### Operating System

Debian 10.11

### How are you running Gitea?

Running `gitea` from systemd on debian10.11

### Database

SQLite

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/zaphbbrox/609acf5cbd9e1d70230f266d4376b589

### Description

After upgrading from a prior version (something like 1.15.x), pushing/pulling became very slow.  
Hope the provided logfile is helpful. If you need more information, please leave a comment.

**Comment by zeripath** [comment]
How are we supposed to interpret the logs that you have provided? 

Is there a push there? When is the push? Where is the pull?

Why haven't you provided router logs?

What is your app.ini?

What is the output from Git?

**Comment by zaphbbrox** [comment]
Sorry, I just filled out the form you provided.  
Also some addition: It's not only push / pull actions, but also sometimes the website actions, like loading the page, etc.  
On the server there's no sign of an overload (low CPU / RAM usage, network's fine).

Here's my app.ini:
```
APP_NAME = Gitea: Git with a cup of tea
RUN_USER = git
RUN_MODE = prod

[oauth2]
JWT_SECRET = SECRET

[security]
INTERNAL_TOKEN     = TOKEN
INSTALL_LOCK       = true
SECRET_KEY         = SECRET_KEY
PASSWORD_HASH_ALGO = pbkdf2

[database]
DB_TYPE  = sqlite3
HOST     = 127.0.0.1:3306
NAME     = gitea
USER     = gitea
PASSWD   = 
SCHEMA   = 
SSL_MODE = disable
CHARSET  = utf8
PATH     = /var/lib/gitea/data/gitea.db
LOG_SQL  = false

[repository]
ROOT = /var/lib/gitea/data/gitea-repositories

[server]
ENABLE_PPROF     = true
SSH_DOMAIN       = domain.example
DOMAIN           = domain.example
HTTP_PORT        = 3000
ROOT_URL         = http://domain.example/
DISABLE_SSH      = false
SSH_PORT         = 22
LFS_START_SERVER = true
LFS_CONTENT_PATH = /var/lib/gitea/data/lfs
LFS_JWT_SECRET   = SECRET
OFFLINE_MODE     = false

[mailer]
ENABLED = true
FROM           = gitea@domain.example
MAILER_TYPE    = smtp
HOST           = mail.domain.example:587
IS_TLS_ENABLED = false
USER           = gitea@domain.example
PASSWD         = SECRET

[service]
REGISTER_EMAIL_CONFIRM            = false
ENABLE_NOTIFY_MAIL                = true
DISABLE_REGISTRATION              = false
ALLOW_ONLY_EXTERNAL_REGISTRATION  = false
ENABLE_CAPTCHA                    = false
REQUIRE_SIGNIN_VIEW               = false
DEFAULT_KEEP_EMAIL_PRIVATE        = true
DEFAULT_ALLOW_CREATE_ORGANIZATION = false
DEFAULT_ENABLE_TIMETRACKING       = false
NO_REPLY_ADDRESS                  = git@domain.example

[picture]
DISABLE_GRAVATAR        = true
ENABLE_FEDERATED_AVATAR = false

[openid]
ENABLE_OPENID_SIGNIN = false
ENABLE_OPENID_SIGNUP = false

[session]
PROVIDER = file

[log]
MODE      = file
LEVEL     = debug
ROOT_PATH = /var/lib/gitea/log
ROUTER    = file
FILE_NAME = gitea.log

[repository.upload]
ENABLED = true
ALLOWED_TYPES = */*
FILE_MAX_SIZE = 200
MAX_FILES = 5

[attachment]
MAX_SIZE = 200

[cache]
ENABLED = true
ADAPTER = memcache
HOST    = 127.0.0.1:11211
```

I'll provide the logs in the following comments and try to filter out the push / pull actions.

**Comment by zaphbbrox** [comment]
These should be log entries regarding a push & pull action:

gitea.log:

```
2022/03/11 11:10:58 ...dules/git/command.go:143:RunWithContext() [D] /var/lib/gitea/data/gitea-repositories/fuw/ueliste.git: /usr/bin/git -c credential.helper= -c protocol.version=2 -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= upload-pack --stateless-rpc --advertise-refs .
2022/03/11 11:14:23 ...dules/git/command.go:143:RunWithContext() [D] /var/lib/gitea/data/gitea-repositories/fuw/ueliste.git: /usr/bin/git -c credential.helper= -c protocol.version=2 -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= receive-pack --stateless-rpc --advertise-refs .
```

router.log:

```
2022/03/11 11:10:58 Started GET /user/repo.git/info/refs?service=git-upload-pack for RECEIVER_IP:0
2022/03/11 11:10:58 Completed GET /user/repo.git/info/refs?service=git-upload-pack 401 Unauthorized in 5.110266ms
2022/03/11 11:10:58 Started GET /user/repo.git/info/refs?service=git-upload-pack for RECEIVER_IP:0
2022/03/11 11:10:58 Completed GET /user/repo.git/info/refs?service=git-upload-pack 200 OK in 34.040951ms
2022/03/11 11:10:58 Started POST /user/repo.git/git-upload-pack for RECEIVER_IP:0
2022/03/11 11:10:58 Completed POST /user/repo.git/git-upload-pack 200 OK in 30.708176ms
2022/03/11 11:14:23 Started GET /user/repo.git/info/refs?service=git-receive-pack for RECEIVER_IP:0
2022/03/11 11:14:23 Completed GET /user/repo.git/info/refs?service=git-receive-pack 401 Unauthorized in 7.795547ms
2022/03/11 11:14:23 Started GET /user/repo.git/info/refs?service=git-receive-pack for RECEIVER_IP:0
2022/03/11 11:14:23 Completed GET /user/repo.git/info/refs?service=git-receive-pack 200 OK in 34.476757ms
```

`git pull` only takes some time, but hasn't any other output than before:  
```
warning: Umleitung nach https://domain.example/user/repo.git/
Bereits aktuell.
```

**Comment by zaphbbrox** [comment]
Seems to be solved with version 1.6.4.
If the issue reappears, I'll reopen this issue.
